### PR TITLE
refactor: implement canvas, remove font

### PR DESCRIPTION
**Pull request by lpnh**
You can test it running the [available template](https://github.com/cadin-org/love-tmpl).

Four main changes:

- Remove the font: it now relies on the current one or default if none was set
- Use the appropriate `canvas` instead of `rectangle`
- Use the `coloredtext` table instead of string
- Use `clear` instead of `setColor` to handle colors

The downside is that I ended up using an OOP approach. Nonetheless, I have a feeling that there's still room to improve. But sooner or later we will find a better solution.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -1,18 +1,5 @@
 local M = {}
```
Comment (vinTibo) [inline init.lua:2]: Add canvas, x, y as a global variable of the module

local canvas
local x, y

File: init.lua [hunk 2]
```diff
@@ -28,43 +15,42 @@ local function splash_colored_text(r, g, b, text)
...
+  love.graphics.printf(colored_text, x, y, text_width, 'center')
   -- TODO: Add fade in/fade out effect
 end
 
-local function draw_splash_screen(r, g, b, text)
-  local x, y, w, h = middle_screen_pos()
-
-  love.graphics.setColor(love.math.colorFromBytes(30, 30, 46))
-  love.graphics.rectangle('fill', x, y, w, h, 12)
+function M.new(self, width, height)
```
Comment (vinTibo) [inline init.lua]: Remove 'self'

File: init.lua [hunk 3]
```diff
@@ -28,43 +15,42 @@ local function splash_colored_text(r, g, b, text)
...
-    splash_colored_text(r, g, b, text)
-  end
-end
+  local x = x_center - (width / 2)
+  local y = y_center - (height / 2)
 
-function M.load()
-  local r, g, b = love.math.colorFromBytes(24, 24, 37)
-  love.graphics.setBackgroundColor(r, g, b)
+  local canvas = love.graphics.newCanvas(width, height)
```
Comment (vinTibo) [inline init.lua]: Remove 'local'

File: init.lua [hunk 4]
```diff
@@ -28,43 +15,42 @@ local function splash_colored_text(r, g, b, text)
...
+  love.graphics.clear(love.math.colorFromBytes(30, 30, 46))
+  love.graphics.setCanvas()
 
-  -- Assumes the following path on the root
-  local font_asset_path = 'assets/fonts/FiraMono-Medium.ttf'
-  local fira_mono = love.graphics.newFont(font_asset_path, 48)
-  love.graphics.setFont(fira_mono)
+  self.canvas = canvas
+  self.x = x
+  self.y = y
```
Comment (vinTibo) [inline init.lua]: Remove this lines

File: init.lua [hunk 5]
```diff
@@ -28,43 +15,42 @@ local function splash_colored_text(r, g, b, text)
...
 end
 
-function M.draw(time)
-  draw_splash_screen()
+function M.start(self, time)
+  love.graphics.draw(self.canvas, self.x, self.y)
```
Comment (vinTibo) [inline init.lua]: Remove  'self'

**Review by vinTibo — APPROVED** [review]
If you want to avoid OOP, you can make the changes that I comment on the code

**Comment by lpnh** [comment]
Isn't it the same thing but without encapsulation?

**Comment by lpnh** [comment]
I've applied your suggestion, se we no longer need to pass the canvas to create and access its properties. However, I think the real "issue" is that we are using the `graphics.draw` function as a way to clean the canvas. Instead of drawing it once and "cleaning it" when needed, we are redrawing it each time. Unfortunately, I haven't found a better way to achieve this yet.

**Pull request merged**